test/spec/clinic/scheduleListController.spec.js

```diff
@@ -1,29 +1,78 @@
-describe('PatientCommonController', function () {
+describe('ScheduleListController', function () {
 
-  var scope, q, controller, observationsService, commonService;
+  var scope, $rootScope, $controller, controller, observationService, commonService, visitService, cohortService;
 
-  beforeEach(inject(function (_$controller_, _$rootScope_, _observationsService_, _commonService_)  {
-      scope =  $rootScope.$new();
-      controller = _$controller_;
-      observationsService = _observationsService_;
-      commonService = _commonService_;
-  }));
+  var visits = [];
 
+  beforeEach(module('clinic'));
+
+  //TODO: Finalize the test after observation Service
   describe('init', function() {
-      beforeEach(function () {
-        spyOn(observationsService, 'getObs').and.callFake(function() {
-            return $q(function (resolve) {
+        beforeEach(inject(function (_$q_, _$controller_, _$rootScope_, _commonService_, _visitService_, _observationsService_, _cohortService_, _$httpBackend_)  {
+          $q = _$q_;
+          $rootScope = _$rootScope_;
+          scope = _$rootScope_.$new();
+          $controller = _$controller_;
+          observationService = _observationsService_;
+          commonService = _commonService_;
+          visitService = _visitService_;
+          cohortService= _cohortService_;
+          $httpBackend = _$httpBackend_;
+        }));
 
-            });
-        });
+        beforeEach(function () {
+          var currentProvider = {
+            uuid: '8d4a4488-c2cc-11de-8d13-0010c6dffd0f'
+
+          }
+          $rootScope.currentProvider = currentProvider;
+
+          spyOn(observationService, 'getObs').and.callFake(function() {
+              return $q(function (resolve) {
+                return resolve([]);
+              });
+          });
+
+          spyOn(visitService, 'search').and.callFake(function() {
+              return $q(function(resolve) {
+                return resolve(visits);
+              });
+          });
+
+          controller = $controller('ScheduleListController',  {
+            $scope: scope,
+            $rootScope: $rootScope,
+            observationService: observationService,
+            visitService: visitService
+          });
+
+          // $scope.$digest();
 
-        it('should call the patient last consultations', function () {
-            $rootScope.$apply();
-            scope.getLastConsultationAndVisit();
-            expect(observationsService.getObs).toHaveBeenCalled();
         });
+          it('should call the patient last consultations', function () {
+
+
+
+              var uuid = 'd00c4b2e-7c13-4d05-9f87-8a9c4b07ad52?providerUuid=8d4a4488-c2cc-11de-8d13-0010c6dffd0f';
+              var currentProvider = {
+                uuid: '8d4a4488-c2cc-11de-8d13-0010c6dffd0f'
+
+              }
+              $rootScope.currentProvider = currentProvider;
+              $httpBackend.whenGET("/poc_config/openmrs/i18n/common/locale_en.json").respond({});
```

Comment:
We already know how to solve this. See [docs](https://angular-translate.github.io/docs/#/guide/22_unit-testing-with-angular-translate#unit-testing-with-angular-translate_the-solution) and [patientCommonController.spec.js](https://github.com/esaude/esaude-emr-poc/blob/master/test/spec/registration/patientCommonController.spec.js#L2).